src/forcemove.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#include "move.h"
+#include "energy.h"
+#include "units.h"
+
+namespace Faunus {
+
+namespace Move {
+
+/**
+ * @brief Base class for dynamics integrators
+ *
+ */
+class IntegratorBase {
+  protected:
+    Space &spc;
+    Energy::Energybase &energy;
+    IntegratorBase(Space &spc, Energy::Energybase &energy);
+    virtual ~IntegratorBase() = default;
+
+  public:
+    /**
+     * @brief Move particles by one time step and update their positions, velocities and acting forces
+     * @param velocities
+     * @param forces
+     */
+    virtual void step(PointVector &velocities, PointVector &forces) = 0; // todo shall we return real time progress?
+    virtual void from_json(const json &j) = 0;
+    virtual void to_json(json &j) const = 0;
+};
+
+void from_json(const json &j, IntegratorBase &i);
+void to_json(json &j, const IntegratorBase &i);
+
+/**
+ * @brief Symmetric Langevin velocity-Verlet method (BAOAB)
+ *
+ * The integrator can conduct normal velocity-Verlet, when gamma = 0.
+ */
+class LangevinVelocityVerlet : public IntegratorBase {
+    float dt;    //!< integration time step (picoseconds)
+    float gamma; //!< friction coefficient (inverse picoseconds)
+
+    //! increment particle's position in an A semi-step
+    inline Point positionIncrement(const Point& velocity);
+    //! increment particle's velocity in a B semi-step
+    inline Point velocityIncrement(const Point& force, const double mass);
+    //! apply fluctuation-dissipation to particle's velocity by solving of Ornstein-Uhlenbeck process in an O-step
+    inline Point velocityFluctuationDissipation(const Point& velocity, const double mass);
+  public:
+    LangevinVelocityVerlet(Space &spc, Energy::Energybase &energy, float dt = 1.0, float gamma = 0.0);
+    LangevinVelocityVerlet(Space &spc, Energy::Energybase &energy, const json &j);
+    void step(PointVector &velocities, PointVector &forces) override;
+    void from_json(const json &j) override;
+    void to_json(json &j) const override;
+};
+
+void from_json(const json &j, IntegratorBase &i);
+void to_json(json &j, const IntegratorBase &i);
+
+/**
+ * @brief Base class for force moves, e.g., molecular dynamics or Langevin dynamics.
+ *
+ * Orchestrate execution of integrators, thermostats, etc. Store vectors for velocities and forces, which are not part
+ * of the particle vector in the space.
+ */
+class ForceMoveBase : public Movebase {
+  protected:
+    unsigned int nsteps; //!< number of integration steps to perform during a single MC move
+    Space &spc;          //!< space with particles and their positions
+    std::shared_ptr<IntegratorBase> integrator;
+    PointVector velocities, forces; //!< velocities and forces; the vector index corresponds to the particle index
```

Comment:
Add a way to load/save velocities from disk? It could be a function that takes a json formatted stream and simply looks for the keys `velocities` and, possibly, `forces`. We could then either use a separate file, or read from / append to the existing `state.json` file.

Reply:
I will create a ticket to resolve this as a separate issue.